### PR TITLE
Added stop button for hello audio worklet example

### PR DESCRIPTION
**Pull request by duojet2ez**
I added a stop button to the hello audio worklet example enabling the user to stop and start the sound. 

Specifically, I used audioContext.suspend() to allow the user to pause the sound with a stop button. And I added an isModuleLoaded check to avoid redundantly calling addModule. Also added isPlaying to toggle the playing state.

### CODE REVIEW

**Comment by duojet2ez** [comment]
@hoch this is the single example version of my previous PR. Just to update you on getting a test page for quick and easy inspection of the PR, I started looking into this again yesterday and it does seem possible to do with github pages + github actions. I will continue researching this over the course of the week and provide updates.

File: src/audio-worklet/basic/hello-audio-worklet/main.js [hunk 1]
```diff
@@ -3,24 +3,50 @@
...
-  const oscillator = new OscillatorNode(context);
-  const bypasser = new AudioWorkletNode(context, 'bypass-processor');
-  oscillator.connect(bypasser).connect(context.destination);
-  oscillator.start();
+  if (!isModuleLoaded) {
+    await context.audioWorklet.addModule('bypass-processor.js');
+    oscillatorNode = new OscillatorNode(context);
+    const bypasser = new AudioWorkletNode(context, 'bypass-processor');
+    oscillatorNode.connect(bypasser).connect(context.destination);
+    oscillatorNode.start();
```
Comment (hoch) [inline src/audio-worklet/basic/hello-audio-worklet/main.js:16]: I think L13~L16 can be extracted to a different method.

`isModuleLoaded` should guard around the `addModule` method, nothing else.

File: src/audio-worklet/basic/hello-audio-worklet/main.js [hunk 2]
```diff
@@ -3,24 +3,50 @@
...
   buttonEl.disabled = false;
+
   buttonEl.addEventListener('click', async () => {
-    await startAudio(audioContext);
-    audioContext.resume();
-    buttonEl.disabled = true;
-    buttonEl.textContent = 'Playing...';
-  }, false);
+    if (!isPlaying) {
+      await startAudio(audioContext);
```
Comment (hoch) [inline src/audio-worklet/basic/hello-audio-worklet/main.js:29]: You can call `audioContext.resume()` here.

File: src/audio-worklet/basic/hello-audio-worklet/main.js [hunk 3]
```diff
@@ -3,24 +3,50 @@
...
-  oscillator.connect(bypasser).connect(context.destination);
-  oscillator.start();
+  if (!isModuleLoaded) {
+    await context.audioWorklet.addModule('bypass-processor.js');
+    oscillatorNode = new OscillatorNode(context);
+    const bypasser = new AudioWorkletNode(context, 'bypass-processor');
+    oscillatorNode.connect(bypasser).connect(context.destination);
+    oscillatorNode.start();
+    isModuleLoaded = true;
+  } else context.resume();
```
Comment (hoch) [inline src/audio-worklet/basic/hello-audio-worklet/main.js:18]: To make the logic clear, please keep this call within the user interaction callback (`click`) below.

File: src/audio-worklet/basic/hello-audio-worklet/main.js [hunk 4]
```diff
@@ -3,24 +3,50 @@
 // found in the LICENSE file.
 
 const audioContext = new AudioContext();
+let isModuleLoaded = false;
```
Comment (hoch) [inline src/audio-worklet/basic/hello-audio-worklet/main.js:6]: You will probably need two different booleans:

`isModuleLoaded`: to guard `addModule()` call.
`isGraphReady`: to guard the node construction/connection logic.

Then the actual `click` event will only care about the context's `resume()` and `suspend()` call.

I believe this is the most efficient way.

File: src/audio-worklet/basic/hello-audio-worklet/main.js [hunk 5]
```diff
@@ -3,24 +3,50 @@
...
+      buttonEl.classList.remove('start-button');
+    } else {
+      audioContext.suspend();
+      isPlaying = false;
+      buttonEl.textContent = 'START';
+      buttonEl.classList.add('start-button');
+    }
+  });
+
+  buttonEl.addEventListener('mouseenter', () => {
```
Comment (hoch) [inline src/audio-worklet/basic/hello-audio-worklet/main.js:41]: I believe L41~L51 is not necessary. Hovering action is not visible on the mobile platform either.

**Review by hoch — CHANGES_REQUESTED** [review]
Thanks for your patience, and I am hopeful that we can land this pretty soon.

**Comment by duojet2ez** [comment]
Thanks for the feedback!! 

- I extracted the following lines mentioned and put them in a loadGraph function:

  oscillatorNode = new OscillatorNode(context);
  const bypasser = new AudioWorkletNode(context, 'bypass-processor');
  oscillatorNode.connect(bypasser).connect(context.destination);
  oscillatorNode.start();

- created a bool isGraphReady to serve as a guard for graph connections in startAudio function
- removed unnecessary lines 41-51
- audioContext.resume() placed in click event 
- Click event should now only care about resume and suspend

**Review by hoch — APPROVED** [review]
LGTM

It looks like there's a conflict, so once you resolve it I'll merge the PR.

**Comment by duojet2ez** [comment]
Cool I think that should resolve the conflict!

**Pull request merged**